src/react/projects/spark-core-react/src/SprkCard/components/SprkCardTeaser/SprkCardTeaser.js

```diff
@@ -0,0 +1,267 @@
+    const MediaImage = () => {
+      if (linkType === 'router-link') {
+        return (
+          <Link
+            to={href}
+            className="sprk-o-Stack__item"
+            data-analytics={mediaLinkAnalytics}
+          >
+            <img
+              className={cx({
+                'sprk-c-Card__media': !titleFirst,
+              })}
+              alt={imgAlt}
+              src={imgSrc}
+            />
+          </Link>
+        );
+      }
+      return (
+        <a
+          href={href}
+          className="sprk-o-Stack__item"
+          data-analytics={mediaLinkAnalytics}
+        >
+          <img
+            className={cx({
+              'sprk-c-Card__media': !titleFirst,
+            })}
+            alt={imgAlt}
+            src={imgSrc}
+          />
+        </a>
+      );
+    };
+
+    const MediaIcon = () => {
+      if (linkType === 'router-link') {
+        return (
+          <Link
+            to={href}
+            className="sprk-o-Stack__item sprk-u-AbsoluteCenter"
+            data-analytics={mediaLinkAnalytics}
+          >
+            <SprkIcon
+              iconName={mediaIconName}
+              additionalClasses={additionalMediaIconClasses}
+            />
+          </Link>
+        );
+      }
+
+      return (
+        <a
+          href={href}
+          className="sprk-o-Stack__item sprk-u-AbsoluteCenter"
+          data-analytics={mediaLinkAnalytics}
+        >
+          <SprkIcon
+            iconName={mediaIconName}
+            additionalClasses={additionalMediaIconClasses}
+          />
+        </a>
+      );
+    };
+
+    if (mediaType === 'img') {
+      return <MediaImage />;
+    }
+
+    return <MediaIcon />;
+  };
+
+  const CallToAction = () => {
+    const GetCtaContent = () => {
+      const CtaLink = () => {
+        if (linkType === 'router-link') {
+          return (
+            <Link
+              to={href}
+              className={cx('sprk-b-Link', additionalCtaClasses, {
+                'sprk-b-Link--simple sprk-b-Link--has-icon': ctaIcon,
+              })}
+            >
+              {ctaText}
+              {ctaIcon && (
+                <SprkIcon
+                  iconName={ctaIcon}
+                  additionalClasses={additionalCtaIconClasses}
+                />
+              )}
+            </Link>
+          );
+        }
+
+        return (
+          <SprkLink
+            variant={ctaIcon ? 'has-icon' : 'base'}
+            additionalClasses={additionalCtaClasses}
+            data-analytics={ctaAnalytics}
+            to={href}
+          >
+            {ctaText}
+            {ctaIcon && (
+              <SprkIcon
+                iconName={ctaIcon}
+                additionalClasses={additionalCtaIconClasses}
+              />
+            )}
+          </SprkLink>
+        );
+      };
+
+      const CtaButton = () => {
+        if (linkType === 'router-link') {
+          return (
+            <Link to={href} className="sprk-c-Button">
```

Comment:
this should use the element pattern like in masthead, doing it this way binds to react-router